src/controllers/csi/provisioner/controller.go

```diff
@@ -258,6 +266,24 @@ func (provisioner *OneAgentProvisioner) getAgentTenantToken(ctx context.Context,
 	return string(token), nil
 }
 
+func (provisioner *OneAgentProvisioner) getAgentProxy(ctx context.Context, dk *dynatracev1beta1.DynaKube) (string, error) {
+	if dk.Spec.Proxy.Value != "" {
+		return dk.Spec.Proxy.Value, nil
+	}
+	query := kubeobjects.NewSecretQuery(ctx, provisioner.client, provisioner.apiReader, log)
+	secret, err := query.Get(types.NamespacedName{Namespace: dk.Namespace, Name: dk.Spec.Proxy.ValueFrom})
+	if err != nil {
+		return "", errors.Wrapf(err, "Proxy secret %s/%s not found", dk.Namespace, dk.Spec.Proxy.ValueFrom)
```

Comment:
just use `dk.Proxy()`, it does the same :D (https://github.com/Dynatrace/dynatrace-operator/blob/3bee602f0e96cb80f55ee2b2193e4adcfa4a6a67/src/api/v1beta1/proxy.go#L33)